lib/scoped/find.js

```diff
@@ -7,7 +7,9 @@ function find (type, api, objectsOrIds) {
 
   .then(function (storedObjects) {
     if (Array.isArray(storedObjects)) {
-      var typedObjects = storedObjects.filter(typeFilter.bind(null, type))
+      var typedObjects = storedObjects.map(function (storedObject) {
+        return typeFilter(type, storedObject) ? storedObject : new Error('Item not found')
```

Comment:
I think best would be to throw a similar error as we do here:
https://github.com/hoodiehq/pouchdb-hoodie-api/blob/master/helpers/find-one.js#L14-L16

Maybe instead of `Object with id "' + id + '" is missing` we can say `Object with type "' + type + '" and id "' + id + '" is missing`